app/src/ui/keyboardlayoutprovider.cpp

```diff
@@ -117,3 +119,55 @@ QVariantList KeyboardLayoutProvider::row6() const
 {
     return m_row6;
 }
+
+int KeyboardLayoutProvider::repeatInterval() const
+{
+    return m_repeatInterval;
+}
+
+void KeyboardLayoutProvider::setRepeatInterval(const int &interval)
+{
+    m_repeatInterval = interval;
+
+    saveConfig();
+
+    emit settingsChanged();
+}
+
+bool KeyboardLayoutProvider::feedback() const
+{
+    return m_feedback;
+}
+
+void KeyboardLayoutProvider::setFeedback(const bool &feedback)
+{
+    m_feedback = feedback;
+
+    saveConfig();
+
+    emit settingsChanged();
+}
+
+void KeyboardLayoutProvider::pressFeedback()
+{
+    QFeedbackHapticsEffect::playThemeEffect(QFeedbackEffect::PressWeak);
+}
+
+void KeyboardLayoutProvider::releaseFeedback()
+{
+    QFeedbackHapticsEffect::playThemeEffect(QFeedbackEffect::ReleaseWeak);
+}
+
+void KeyboardLayoutProvider::saveConfig() const
+{
+    QFile config(QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/harbour-sailfishconnect/keyboardlayout.conf");
```

Comment:
Please use QSettings for application-wide settings. Use "RemoteKeyboard" as group. As example see the runInBackground setting in the UI class: 
https://github.com/R1tschY/harbour-sailfishconnect/blob/2d6145c5f084370251d66b5b6a6e8f6cf1223528/app/src/ui.cpp#L49
https://github.com/R1tschY/harbour-sailfishconnect/blob/2d6145c5f084370251d66b5b6a6e8f6cf1223528/app/src/ui.cpp#L138